### ISSUE TITLE
Use relative units for font-size and related lengths

### ISSUE THREAD

**Issue by missmatsuko** [craftcms/cms#6890]
### Description

We should use relative units (e.g. em, rem, %) for font size and related lengths (e.g. line-height).

Setting heights/widths on elements should be avoided as much as possible, allowing the content to determine the element's height instead. If a height/width has to be set, try to use relative units as well, and make sure it updates when font-size or zoom changes up to 200%.

I moved this to 4.0 because it's [less of an issue in modern browsers](https://www.24a11y.com/2019/pixels-vs-relative-units-in-css-why-its-still-a-big-deal/) and will be easier to get right in a new design/build.

References:
- [WCAG 2.0 SC 1.4.4 Resize text](https://www.w3.org/TR/2008/REC-WCAG20-20081211/#visual-audio-contrast-scale)
- [How to meet 1.4.4](https://www.w3.org/WAI/WCAG21/quickref/?versions=2.0#qr-visual-audio-contrast-scale)
- [Understanding 1.4.4](https://www.w3.org/TR/UNDERSTANDING-WCAG20/visual-audio-contrast-scale.html)

### Steps to reproduce
1. Increase body font size to 200% (e.g. 28px) and observe text size and... squishiness.

### Additional info

- Craft version: 3.5.11
